keras_rs/src/losses/pairwise_loss.py

```diff
@@ -0,0 +1,93 @@
+import abc
+from typing import Optional
+
+import keras
+from keras import ops
+
+from keras_rs.src import types
+from keras_rs.src.utils.pairwise_loss_utils import pairwise_comparison
+from keras_rs.src.utils.pairwise_loss_utils import process_loss_call_inputs
+
+
+class PairwiseLoss(keras.losses.Loss):
+    """Base class for pairwise ranking losses.
+
+    Pairwise loss functions are designed for ranking tasks, where the goal is to
+    correctly order items within each list. Any pairwise loss function computes
+    the loss value by comparing pairs of items within each list, penalizing
+    cases where an item with a higher true label has a lower predicted score
+    than an item with a lower true label.
+
+    In order to implement any kind of pairwise loss, override the
+    `pairwise_loss` method.
+    """
+
+    # TODO: Add `temperature`, `lambda_weights`.
```

Comment:
Was this in the original implementation? Can you create a github issue for this?

Reply:
Yep, it's there in the original implementation